File: merge-pdfs.js
const { PDFDocument } = require('pdf-lib');
const fs = require('fs').promises;
const path = require('path');

async function mergePDFs() {
    try {
        console.log('Starting PDF merge...');
        
        // Read the PDF files
        console.log('Reading cover PDF...');
        const coverPdfBytes = await fs.readFile('public/omslag-complete.pdf');
        
        console.log('Reading main book PDF...');
        const mainPdfBytes = await fs.readFile('e-book_weedyourskin.pdf');
        
        // Load PDFs
        console.log('Loading PDFs...');
        const coverPdf = await PDFDocument.load(coverPdfBytes);
        const mainPdf = await PDFDocument.load(mainPdfBytes);
        
        // Create a new PDF
        const mergedPdf = await PDFDocument.create();
        
        // Copy pages from cover
        console.log('Copying cover pages...');
        const coverPages = await mergedPdf.copyPages(coverPdf, coverPdf.getPageIndices());
        coverPages.forEach((page) => mergedPdf.addPage(page));
        
        // Copy pages from main book
        console.log('Copying main book pages...');
        const mainPages = await mergedPdf.copyPages(mainPdf, mainPdf.getPageIndices());
        mainPages.forEach((page) => mergedPdf.addPage(page));
        
        // Save the merged PDF
        console.log('Saving merged PDF...');
        const mergedPdfBytes = await mergedPdf.save();
        
        // Write to file
        await fs.writeFile('merged-ebook.pdf', mergedPdfBytes);
        
        // Get file sizes
        const coverSize = (await fs.stat('public/omslag-complete.pdf')).size / (1024 * 1024);
        const mainSize = (await fs.stat('e-book_weedyourskin.pdf')).size / (1024 * 1024);
        const mergedSize = (await fs.stat('merged-ebook.pdf')).size / (1024 * 1024);
        
        console.log('\nSuccess! Files merged:');
        console.log(`  Cover: ${coverSize.toFixed(1)} MB`);
        console.log(`  Main book: ${mainSize.toFixed(1)} MB`);
        console.log(`  Merged: ${mergedSize.toFixed(1)} MB`);
        
        // Rename files
        console.log('\nBacking up original file...');
        await fs.rename('e-book_weedyourskin.pdf', 'e-book_weedyourskin_backup.pdf');
        await fs.rename('merged-ebook.pdf', 'e-book_weedyourskin.pdf');
        
        console.log('Done! The merged file is now e-book_weedyourskin.pdf');
        console.log('Original file backed up as e-book_weedyourskin_backup.pdf');
        
    } catch (error) {
        console.error('Error merging PDFs:', error);
    }
}

mergePDFs(); 

File: merge-pdfs.py
#!/usr/bin/env python3
import subprocess
import sys

# Install PyPDF2 if not already installed
try:
    import PyPDF2
except ImportError:
    print("Installing PyPDF2...")
    subprocess.check_call([sys.executable, "-m", "pip", "install", "--user", "PyPDF2"])
    import PyPDF2

import os

def merge_pdfs():
    # Check if files exist
    if not os.path.exists('public/omslag-complete.pdf'):
        print("Error: public/omslag-complete.pdf not found!")
        return
    
    if not os.path.exists('e-book_weedyourskin.pdf'):
        print("Error: e-book_weedyourskin.pdf not found!")
        return
    
    print("Merging PDFs...")
    
    # Create PDF merger
    merger = PyPDF2.PdfMerger()
    
    try:
        # Add cover PDF
        print("Adding cover...")
        merger.append('public/omslag-complete.pdf')
        
        # Add main book PDF
        print("Adding main book...")
        merger.append('e-book_weedyourskin.pdf')
        
        # Write merged PDF
        output_file = 'merged-ebook.pdf'
        print(f"Writing merged PDF to {output_file}...")
        merger.write(output_file)
        merger.close()
        
        # Get file sizes
        cover_size = os.path.getsize('public/omslag-complete.pdf') / (1024*1024)
        main_size = os.path.getsize('e-book_weedyourskin.pdf') / (1024*1024)
        merged_size = os.path.getsize(output_file) / (1024*1024)
        
        print(f"\nSuccess! Files merged:")
        print(f"  Cover: {cover_size:.1f} MB")
        print(f"  Main book: {main_size:.1f} MB")
        print(f"  Merged: {merged_size:.1f} MB")
        print(f"\nMerged file saved as: {output_file}")
        
        # Rename files
        print("\nRenaming files...")
        os.rename('e-book_weedyourskin.pdf', 'e-book_weedyourskin_backup.pdf')
        os.rename(output_file, 'e-book_weedyourskin.pdf')
        print("Done! The merged file is now e-book_weedyourskin.pdf")
        
    except Exception as e:
        print(f"Error merging PDFs: {e}")
        merger.close()

if __name__ == "__main__":
    merge_pdfs() 